src/test/ui/methods/issues/issue-90315.rs

```diff
@@ -1,7 +1,54 @@
 fn main() {
-  let arr = &[0,1,2,3];
-  for _i in 0..arr.len().rev() { //~ERROR not an iterator
-     // The above error used to say “the method `rev` exists for type `usize`”.
-     // This regression test ensures it doesn't say that any more.
-  }
+    let arr = &[0, 1, 2, 3];
+    for _i in 0..arr.len().rev() { //~ERROR not an iterator
+        // The above error used to say “the method `rev` exists for type `usize`”.
+        // This regression test ensures it doesn't say that any more.
+    }
+
+    // Test
+    for i in 1..11.rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    let end: usize = 10;
+    for i in 1..end.rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    for i in 1..(end + 1).rev() {
+        //~^ ERROR not an iterator
+        // Suggest add the missing parentheses for range
+        println!("{i}");
+    }
+
+    if 1..(end + 1).is_empty() {
+        //~^ ERROR not an iterator
+        //~^^ ERROR mismatched types [E0308]
+        // Suggest add the missing parentheses for range
+        println!("empty");
+    }
+
+    if 1..(end + 1).is_sorted() {
+        //~^ ERROR mismatched types [E0308]
+        //~^^ ERROR `usize` is not an iterator [E0599]
+        // Suggest add the missing parentheses for range
+        println!("empty");
+    }
+
+    if 1..end.error_method() {
+        //~^ ERROR no method named `error_method`
+        //~^^ ERROR mismatched types [E0308]
+        // Won't suggest
+    }
+
+    let _res: i32 = 3..6.take(1).sum();
+    //~^ ERROR `{integer}` is not an iterator [E0599]
+    //~^^ ERROR mismatched types [E0308]
+    let _sum: i32 = 3..6.sum();
+    //~^ ERROR `{integer}` is not an iterator [E0599]
+    //~^^ ERROR mismatched types [E0308]
 }
```

Comment:
Is it worth it to add more tests regarding inclusive range (`..=`) and where the beginning of the range is a variable which is not tested here? (Possibly where both are variables too). Not sure if that will be too much.

Reply:
Sounds great, will add one..